Code/Tools/ProjectManager/Source/DownloadController.cpp

```diff
@@ -37,40 +36,45 @@ namespace O3DE::ProjectManager
         m_workerThread.wait();
     }
 
-    void DownloadController::AddGemDownload(const QString& gemName)
+    void DownloadController::AddObjectDownload(const QString& objectName, DownloadObjectType objectType)
     {
-        m_gemNames.push_back(gemName);
-        emit GemDownloadAdded(gemName);
+        m_objects.push_back({ objectName, objectType });
+        emit ObjectDownloadAdded(objectName, objectType);
 
-        if (m_gemNames.size() == 1)
+        if (m_objects.size() == 1)
         {
-            m_worker->SetGemToDownload(m_gemNames.front(), false);
+            m_worker->SetObjectToDownload(m_objects.front().m_objectName, objectType, false);
             m_workerThread.start();
         }
     }
 
-    void DownloadController::CancelGemDownload(const QString& gemName)
+    void DownloadController::CancelObjectDownload(const QString& objectName, DownloadObjectType objectType)
     {
-        auto findResult = AZStd::find(m_gemNames.begin(), m_gemNames.end(), gemName);
+        auto findResult = AZStd::ranges::find_if(m_objects,
+            [objectName, objectType](const DownloadableObject& object)
+            {
+                return (object.m_objectType == objectType && object.m_objectName == objectName);
+            });
 
-        if (findResult != m_gemNames.end())
+        if (findResult != m_objects.end())
         {
-            if (findResult == m_gemNames.begin())
+            if (findResult == m_objects.begin())
             {
-                // HandleResults will remove the gem upon cancelling
+                // HandleResults will remove the object upon cancelling
                 PythonBindingsInterface::Get()->CancelDownload();
             }
             else
             {
-                m_gemNames.erase(findResult);
-                emit GemDownloadRemoved(gemName);
+                m_objects.erase(findResult);
+                emit ObjectDownloadRemoved(objectName, objectType);
             }
         }
     }
 
     void DownloadController::UpdateUIProgress(int bytesDownloaded, int totalBytes)
     {
-        emit GemDownloadProgress(m_gemNames.front(), bytesDownloaded, totalBytes);
+        const DownloadableObject& downloadableObject = m_objects.front();
+        emit ObjectDownloadProgress(downloadableObject.m_objectName, downloadableObject.m_objectType, bytesDownloaded, totalBytes);
```

Comment:
This should check that m_objects is not empty
```suggestion
        if (!m_objects.empty())
        {
            const DownloadableObject& downloadableObject = m_objects.front();
            emit ObjectDownloadProgress(downloadableObject.m_objectName, downloadableObject.m_objectType, bytesDownloaded, totalBytes);
        }
```